File: C++/Problems/Problems/problem_1344.cpp
#include "problem_1344.h"

bool problem_1344::test() {
    int hour = 12;
    int minutes = 30;

    double expected = 165;

    auto result = angleClock(hour, minutes);

    std::cout << result << std::endl;

    return std::abs(result - expected) < 1e-5;
}

double problem_1344::angleClock(int hour, int minutes) {
    double angle = std::abs(hour * 30 + minutes / 2.0 - minutes * 6);
    return angle < 180 ? angle : 360 - angle;
}

File: Python/Problems/Problems/problem1344.py
from itertools import repeat

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem1344(Problem):
    def test(self) -> bool:
        hour = 12
        minutes = 30

        expected = 165

        result = self.angleClock(hour, minutes)

        print(result)

        return abs(result - expected) < 1e-5

    def angleClock(self, hour: int, minutes: int) -> float:
        angle = abs(hour * 30 + minutes / 2.0 - minutes * 6)
        return angle if angle < 180 else 360 - angle